stable/aws-calico/templates/daemon-set.yaml

```diff
@@ -19,6 +19,15 @@ spec:
         app.kubernetes.io/name: "{{ include "aws-calico.fullname" . }}-node"
     spec:
       priorityClassName: system-node-critical
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                - key: "eks.amazonaws.com/compute-type"
+                  operator: NotIn
+                  values:
+                    - fargate
```

Comment:
Thanks, looks good. We should probably add checks for OS and Arch as well, since the default Calico image will only work on Linux on amd64. That can be a separate PR.